src/Kunstmaan/MediaBundle/Controller/MediaController.php

```diff
@@ -472,4 +493,46 @@ public function moveMedia(Request $request)
 
         return new JsonResponse();
     }
+
+    /**
+     * @Route("/bulk-move", name="KunstmaanMediaBundle_media_bulk_move")
+     *
+     * @param Request $request
+     *
+     * @return JsonResponse|Response
+     * @throws \Doctrine\DBAL\DBALException
+     */
+    public function bulkMoveAction(Request $request)
+    {
+        $em = $this->getDoctrine()->getManager();
+        $mediaRepo = $em->getRepository('KunstmaanMediaBundle:Media');
+        $form = $this->createForm(BulkMoveMediaType::class);
+
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            /** @var Folder $folder */
+            $folder = $form->getData()['folder'];
+            $mediaIds = explode(',', $form->getData()['media']);
+
+            foreach ($mediaIds as $mediaId) {
+                $mediaRepo->moveToFolder($mediaId, $folder->getId());
```

Comment:
This is still executing queries in a loop. I was thinking more like:

```php
$mediaRepo->createQueryBuilder('media')
    ->update()
    ->set('media.folder', $folder->getId())
    ->where('media.id in (:mediaIds)')
    ->setParameter('mediaIds', $mediaIds)
    ->getQuery()
    ->execute();
```

Reply:
True, didn't see that ;) I've changed it now.